packages/integration-tests/sfnStep.js

```diff
@@ -122,22 +122,41 @@ class SfnStep {
       return null;
     }
 
-    const scheduleEvent = stepExecutions[0]['scheduleEvent'];
+    const scheduleEvent = stepExecutions[0].scheduleEvent;
     const eventWasSuccessful = scheduleEvent.type === this.scheduleSuccessfulEvent;
     if (!eventWasSuccessful) console.log('Schedule event failed');
 
-    const subStepExecutionDetails = scheduleEvent['lambdaFunctionScheduledEventDetails'];
+    const subStepExecutionDetails = scheduleEvent.lambdaFunctionScheduledEventDetails;
     return JSON.parse(subStepExecutionDetails.input);
-  };
+  }
+
+  getSuccessOutput(stepExecution, stepName) {
+    if (stepExecution.completeEvent.type !== this.successEvent) {
+      console.log(`Step ${stepName} did not complete successfully, as expected.`);
+      return null;
+    }
+    const completeEventOutput = stepExecution.completeEvent[this.eventDetailsKeys.succeeded];
+    return JSON.parse(completeEventOutput.output.toString());
+  }
+
+  getFailureOutput(stepExecution, stepName) {
+    if (stepExecution.completeEvent.type !== this.failureEvent) {
+      console.log(`Step ${stepName} did not fail, as expected.`);
+      return null;
+    }
+    const completeEventOutput = stepExecution.completeEvent[this.eventDetailsKeys.failed];
+    return completeEventOutput;
+  }
 
   /**
    * Get the output payload from the step, if the step succeeds
    *
    * @param {string} workflowExecutionArn - Arn of the workflow execution
    * @param {string} stepName - name of the step
+   * @param {string} eventType - expected type of event, should be 'success' or 'failure'
    * @returns {Object} object containing the payload, null if error
    */
-  async getStepOutput(workflowExecutionArn, stepName) {
+  async getStepOutput(workflowExecutionArn, stepName, eventType = 'success') {
```

Comment:
If I understand correctly, in the for `eventType` === 'failure', we're relying on the last execution to be the failed one?  Is that a valid assumption?  we filter for only successful executions on L169.  Is there a case where this wont work?  I guess these are test helpers, so maybe?

Reply:
Good call, I didn't notice that filter, I've updated the conditional and comment